--- a/net/minecraft/command/impl/KillCommand.java
+++ b/net/minecraft/command/impl/KillCommand.java
@@ -7,6 +_,9 @@
 import net.minecraft.command.Commands;
 import net.minecraft.command.arguments.EntityArgument;
 import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.util.DamageSource;
 import net.minecraft.util.text.TranslationTextComponent;
 
 public class KillCommand {
@@ -22,7 +_,11 @@
 
    private static int func_198519_a(CommandSource p_198519_0_, Collection<? extends Entity> p_198519_1_) {
       for(Entity entity : p_198519_1_) {
-         entity.func_174812_G();
+         if (entity instanceof PlayerEntity) {
+            ((PlayerEntity) entity).func_70645_a(DamageSource.field_76380_i); // NeptuneX - Fix /kill command not working.
+         } else {
+            entity.func_174812_G();
+         }
       }
 
       if (p_198519_1_.size() == 1) {
